### ISSUE TITLE
Alt text on images opening from lightbox links

### ISSUE THREAD

**Issue by FayCross** [thexerteproject/xerteonlinetoolkits#1096]
When a link is created that opens an image in a lightbox, there is no way to add alt text for that image. There needs to be a new field in the link dialog:

![imgLink2](https://user-images.githubusercontent.com/8926470/166670980-cfaef132-9beb-4bd2-8550-e3dd07e7a3ba.PNG)

**Comment by FayCross** [comment]
@JohnSmith-LT I think that works fine but I'll keep open until Ron has tested. Thanks

**Comment by JohnSmith-LT** [comment]
Thanks @FayCross

**Comment by ronm123** [comment]
@JohnSmith-LT @FayCross have just tested this. John's fix is definitely an improvement given the Alt text option when choosing image from the menu didn't exist before and I can't see this breaking anything however a couple of things I noticed in testing...

Firstly if I wanted an image to open in a lightbox I would rarely link to an image in this way e.g. from a text link. I would typically just add the image to the text area at a reduced percentage, which then obviously provides an alt text option by default and opens the same image in a lightbox by default. I would also usually add the lightbox optional property and change the caption option to above or below image instead of none.

However I can see reasons authors might want to just use a text link to open an image in lightbox and I guess Fay added the issue based on real world examples too. So this raises the question if an author want to use a text link to an image in this way and then adds the alt text should we/can we also check if the lightbox optional property has been added to show captions and if so show the caption like it does when an image link is used instead?

Also if an author uses a text link to an image and chooses Lightbox but doesn't change the Type menu to Image and leaves it as the default of iFrame then that link to the image still works but obviously doesn't have the alt text and doesn't resize the image to fit the lightbox either. So is there a way to add help text/tooltip? e.g.
![image](https://user-images.githubusercontent.com/5167146/167644575-be2b043c-f132-4a64-b088-50f6060813d0.png)


Finally this must be something I just haven't noticed before and is just a bit odd - when I click around in the link window I get a flashing cursor in different places depending on where I click e.g. see circled cursor above. Not a major problem just a bit odd?

**Comment by FayCross** [comment]
@ronm123 good point about grabbing the optional property info about whether a caption should be used - that would be good to have too. I also agree about needing to make it clearer that you need to change from iframe to image to make sure the image is responsive etc. I initially forgot this last week when testing and it took a while to work out why it wasn't working as expected.

**Comment by JohnSmith-LT** [comment]
Rolled codebase back to 29th April and that strange flashing cursor still appears so it's not something introduced by new code @ronm123 but I've never noticed it before either... it happens all over the plugin. I will take a look to see if I can figure out why it is happening...

![StrangeCursor](https://user-images.githubusercontent.com/3907362/168383066-6dc18e56-f208-491d-9f9f-271daee2c4c5.gif)

**Comment by JohnSmith-LT** [comment]
Figured out the flashing cursor @ronm123

Presumably, like me, you have accidentally hit F7 or have turned on this accessibility feature in Chrome:

![image](https://user-images.githubusercontent.com/3907362/168424861-68724011-e921-4bd4-bd2a-7eb5929f60e3.png)

**Comment by JohnSmith-LT** [comment]
Let's keep this open until I can get the Tip: in screen in the box as I think that's a very good idea.

**Comment by JohnSmith-LT** [comment]
I've added this tooltip now to the plugin so if you are both happy with this @FayCross and @ronm123 then feel free to close now.

**Comment by FayCross** [comment]
@JohnSmith-LT That's great - works really well.
